### PR TITLE
feat: Adds new script for deploying/updating sdks

### PR DESCRIPTION
**Pull request by dckesler**
### Description

Adds a new script that makes deploying the sdks much easier.

### Other changes

Updated a few dependencies including Typescript which then made me make a few chances in contractkit for error handling.

### Tested

Ran the script several times using the dry-run option and worked out all the edge case issues.

### Backwards compatibility

Yes backwards compatible. Nothing is changing with the actual packages.

### Documentation

Documentation was added to the script itself.

### CODE REVIEW

File: packages/sdk/contractkit/src/identity/claims/verify.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export const verifyAccountClaim = async (
   let metadata: IdentityMetadataWrapper
   try {
     metadata = await IdentityMetadataWrapper.fetchFromURL(kit, metadataURL, tries)
-  } catch (error) {
+  } catch (error: any) {
```
Comment (dckesler) [inline packages/sdk/contractkit/src/identity/claims/verify.ts]: There's a lot of this. The newer version of Typescript changed the type for the catch block parameter to be "unknown" but you get the previous behavior by adding `: any`.

File: packages/sdk/wallets/wallet-ledger/src/ledger-wallet.test.ts [hunk 2]
```diff
@@ -179,7 +179,7 @@ describe('LedgerWallet class', () => {
         // Use physical ledger if present
         // Recreation of the connection will fail, therefore we use a single object
         if (!hardwareWallet) {
-          const transport: Transport = await TransportNodeHid.open('')
+          const transport = await TransportNodeHid.open('')
```
Comment (dckesler) [inline packages/sdk/wallets/wallet-ledger/src/ledger-wallet.test.ts]: This `Transport` type didn't seem to exist and was breaking the builds.

File: packages/sdk/wallets/wallet-walletconnect/package.json [hunk 3]
```diff
@@ -30,6 +30,7 @@
     "@celo/wallet-remote": "1.2.5-dev",
     "@walletconnect/client": "2.0.0-beta.18",
     "@walletconnect/types": "2.0.0-beta.18",
+    "@walletconnect/utils": "2.0.0-beta.18",
```
Comment (dckesler) [inline packages/sdk/wallets/wallet-walletconnect/package.json]: This is also used in this package. I'm not sure how it wasn't breaking before.

Comment (aaronmgdr) [reply packages/sdk/wallets/wallet-walletconnect/package.json]: sometimes i a dependency includes a sub dependency you can import the sub dep with it worrying.

File: scripts/deploy-sdks.ts [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+  )
+  const sdkNames = sdkJsons.map(({ name }) => name)
+
+  let newVersion: string
+  // Here we create the new `package.json` objects with updated
+  // versions and dependencies.
+  sdkJsons.map((json) => {
+    if (!newVersion) {
+      if (!version) newVersion = removeDevSuffix(json.version)
+      else
```
Comment (aaronmgdr) [inline scripts/deploy-sdks.ts]: nit: but this if else doesnt have {}

File: scripts/deploy-sdks.ts [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+  dependencies: { [key: string]: string }
+  devDependencies: { [key: string]: string }
+}
+
+type Answers = {
+  packages: string[]
+  version: string
+  publish: string
+}
+;(async function () {
```
Comment (nambrot) [inline scripts/deploy-sdks.ts:57]: Optional Nit: This is a pretty massive function, I would recommend to break it down

File: scripts/deploy-sdks.ts [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+
+  const shouldPublish = publish === 'Y' || publish === 'dry-run'
+
+  if (!shouldPublish && !version) {
+    console.error(colors.red('Either a version or --publish must be given'))
+    process.exit()
+  }
+
+  const sdkPackagesFolder = path.join(__dirname, '..', 'packages', 'sdk')
+  const sdkPackagePaths = findPackagePaths(sdkPackagesFolder)
```
Comment (nambrot) [inline scripts/deploy-sdks.ts]: If all these variables are correlated to each other, why not hold a single state variable?

**Review by nambrot — APPROVED** [review]

File: scripts/deploy-sdks.ts (hunk 5 above)
Comment (dckesler) [reply scripts/deploy-sdks.ts:57]: Yeah, you're right. It's just a big async IIFE so that we can write `async/await` so I didn't think much about it.

**Comment by AlexBHarley** [comment]
@dckesler finally got to playing around with this today.

1. It's weird that dry-run npm publishes don't fail if you give an incorrect code. In the payments script I don't even do an NPM publish... maybe worth adding
2. Something seems off when retrying to publish failed packages. It increments all the versions again and then prompts you to publish only the failed ones. I would also expect re runs to ask for the `dry-run` option again, seems like a foot gun otherwise

**Comment by dckesler** [comment]
Oh weird it's not supposed to increment the versions again. It should though save the last publish command. I think I know what's wrong with the incrementing though.

**Pull request merged**